### PR TITLE
VTOL limit min thrust

### PR DESCRIPTION
**Pull request by tumbili**
Prevent low thrust setting which leads to lack of airflow over control surfaces during descent in hover.
Only do this in altitude control mode.

### CODE REVIEW

**Comment by tumbili** [comment]
Not yet fully flight tested, do not merge yet!

**Comment by LorenzMeier** [comment]
@tumbili This branch is great for learning how to rebase. Make sure to ```git fetch origin``` and always use ```origin/master```, not ```master``` as branch reference, so you rebase onto the right one.
https://help.github.com/articles/using-git-rebase/

**Comment by tumbili** [comment]
Cool, thanks!

**Comment by LorenzMeier** [comment]
Re-kicking Hans on this one.

**Comment by LorenzMeier** [comment]
@tumbili Flown?

**Comment by tumbili** [comment]
Function is implemented but not used yet. Want to test tomorrow if this is even needed.

File: src/modules/vtol_att_control/vtol_att_control_main.cpp [hunk 1]
```diff
@@ -666,6 +671,18 @@ void VtolAttitudeControl::calc_tot_airspeed() {
...
+void VtolAttitudeControl::limit_min_thrust() {
+	// filter thrust command
+	_thrust_filt = _params.arsp_lp_gain*(_thrust_filt - _actuators_mc_in.control[3]) + _actuators_mc_in.control[3];
+	// prevent too low airspeeds over control surfaces in altitude control mode
+	if(_airspeed_tot < _params.mc_airspeed_min && _v_control_mode.flag_control_altitude_enabled && _thrust_low_bound < 0.001f) {
+		_thrust_low_bound = _thrust_filt;
+	}
+	if(_actuators_mc_in.control[3] > _thrust_low_bound + 0.05f || !_v_control_mode.flag_control_altitude_enabled) {
+		_thrust_low_bound = 0.0f;
+	}
```
Comment (bansiesta) [inline src/modules/vtol_att_control/vtol_att_control_main.cpp]: I don't understand what exactly you are trying to solve with _thrust_low_bound.
You're basically setting _thrust_low_bound *once* when airspeed is too low and you reset it to 0 when you throttle up in non-altitude mode?

File: src/modules/vtol_att_control/vtol_att_control_main.cpp [hunk 2]
```diff
@@ -778,6 +795,8 @@ void VtolAttitudeControl::task_main()
 
 				// scale pitch control with total airspeed
 				scale_mc_output();
+				// limit minimum thrust to prevent lack of airflow over control surfaces in hover
+				//limit_min_thrust();
```
Comment (bansiesta) [inline src/modules/vtol_att_control/vtol_att_control_main.cpp]: Ah, you're not using it anyway? :)

Comment (tumbili) [inline src/modules/vtol_att_control/vtol_att_control_main.cpp]: Yeah, we can close this PR, I was using it when I flew the VTOL in combination with the mc_pos_controller for attitude control. But now there is a vtol_pos_control module which takes care of all of this. It is much cleaner anyway.

Comment (bansiesta) [inline src/modules/vtol_att_control/vtol_att_control_main.cpp]: :thumbsup:

**Pull request closed without merging**